extensions/podman/src/extension.ts

```diff
@@ -208,22 +216,50 @@ export async function checkDefaultMachine(machines: MachineJSON[]): Promise<void
   // As a last check, let's see if the machine that is running is set by default or not on the CLI.
   // if it isn't, we should prompt the user to set it as default, or else podman CLI commands will not work
   const runningMachine = machines.find(machine => machine.Running);
-  const defaultMachine = machines.find(machine => machine.Default);
+  let defaultMachine = machines.find(machine => machine.Default);
+  if (!defaultMachine) {
+    const defaultConnection = await getDefaultConnection();
+    let defaultConnectionName = defaultConnection?.Name;
+    if (defaultConnectionName.endsWith('-root')) {
+      defaultConnectionName = defaultConnectionName.substring(0, defaultConnectionName.length - 5);
+    }
+    defaultMachine = machines.find(machine => machine.Name === defaultConnectionName);
+
+    if (runningMachine?.Name === defaultConnectionName) {
+      runningMachine.Default = true;
+    }
+  }
 
   if (defaultMachineNotify && defaultMachineMonitor && runningMachine && !runningMachine.Default) {
     // Make sure we do notifyDefault = false so we don't keep notifying the user when this dialog is open.
     defaultMachineMonitor = false;
 
     // Create an information message to ask the user if they wish to set the running machine as default.
     const result = await extensionApi.window.showInformationMessage(
-      `Podman Machine '${runningMachine.Name}' is running but not the default machine (default is '${defaultMachine.Name}'). This will cause podman CLI errors while trying to connect to '${runningMachine.Name}'. Do you want to set it as default?`,
+      `Podman Machine '${runningMachine.Name}' is running but not the default machine ${
+        defaultMachine ? '(default is ' + defaultMachine.Name + ')' : ''
+      }. This will cause podman CLI errors while trying to connect to '${
+        runningMachine.Name
+      }'. Do you want to set it as default?`,
       'Yes',
       'Ignore',
       'Cancel',
     );
     if (result === 'Yes') {
       try {
+        // make it the default to run the info command
         await execPromise(getPodmanCli(), ['system', 'connection', 'default', runningMachine.Name]);
+
+        // check if it's rootful
+        const machineInfoJson = await execPromise(getPodmanCli(), ['machine', 'info', '--format', 'json']);
+        const machineInfo = JSON.parse(machineInfoJson);
+        const filepath = path.join(machineInfo.Host.MachineConfigDir, `${runningMachine.Name}.json`);
+        const machineConfigJson = await readFile(filepath);
```

Comment:
should we check if file exists or that we have an entry ?

Reply:
added a path/content check + moved to its own try/catch in case of other failures